### PR TITLE
fix: don't checkpoint gc in exp delete with no checkpoints [DET-8834]

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Don't create checkpoint GC task if we don't have any checkpoints.

## Test Plan

intg + e2e tests

Create an experiment that will fail and cause checkpoint GC to fail.

```
det e create --config="checkpoint_storage.type=s3" --config="checkpoint_storage.bucket=notarealbucket" examples/tutorials/mnist_pytorch/const.yaml examples/tutorials/mnist_pytorch/ -f
```

Delete experiment after it errors
```
det e delete $exp_id
```

Verify the exp is gone from ```det e```

## Commentary (optional)

Created DET-8836 to follow up about how we can handle failed checkpoint GC better.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: e2e_tests/tests/cluster/test_checkpoints.py [hunk 1]
```diff
@@ -336,6 +336,32 @@ def test_s3_no_creds(secrets: Dict[str, str]) -> None:
...
+
+    # Still able to delete this since it will have no checkpoints meaning no checkpoint gc task.
+    test_session = exp.determined_test_session()
+    bindings.delete_DeleteExperiment(session=test_session, experimentId=exp_id)
+    for _ in range(60):
+        try:
+            print(f"experiment in state {exp.experiment_state(exp_id)} waiting for deleted")
+            time.sleep(1)
+        except api.errors.NotFoundException:
+            return
```
Comment (dzhu) [inline e2e_tests/tests/cluster/test_checkpoints.py:362]: Getting past the loop should be a test failure.

File: e2e_tests/tests/cluster/test_checkpoints.py [hunk 2]
```diff
@@ -336,6 +336,32 @@ def test_s3_no_creds(secrets: Dict[str, str]) -> None:
...
+        conf.fixtures_path("no_op"),
+        None,
+    )
+
+    # Still able to delete this since it will have no checkpoints meaning no checkpoint gc task.
+    test_session = exp.determined_test_session()
+    bindings.delete_DeleteExperiment(session=test_session, experimentId=exp_id)
+    for _ in range(60):
+        try:
+            print(f"experiment in state {exp.experiment_state(exp_id)} waiting for deleted")
```
Comment (dzhu) [inline e2e_tests/tests/cluster/test_checkpoints.py:359]: Probably don't need to print every time. Also, having the check be contained inside the evaluation of the format string is a bit opaque.

File: master/internal/api_experiment_intg_test.go [hunk 3]
```diff
@@ -149,6 +149,29 @@ func TestGetExperimentLabels(t *testing.T) {
...
+	exp := createTestExp(t, api, curUser)
+	_, err := db.Bun().NewUpdate().Table("experiments").
+		Set("state = ?", model.CompletedState).
+		Where("id = ?", exp.ID).Exec(ctx)
+	require.NoError(t, err)
+
+	_, err = api.DeleteExperiment(ctx, &apiv1.DeleteExperimentRequest{ExperimentId: int32(exp.ID)})
+	require.NoError(t, err)
+
+	// Delete is aync so we need to retry until it completes.
```
Comment (dzhu) [inline master/internal/api_experiment_intg_test.go:163]:
```suggestion
	// Delete is async, so we need to retry until it completes.
```

File: master/internal/api_user_intg_test.go [hunk 4]
```diff
@@ -52,6 +52,10 @@ func setupAPITest(t *testing.T, pgdb *db.PgDB) (*apiServer, model.User, context.
 			system = actor.NewSystem("mock")
 			ref, _ := system.ActorOf(sproto.K8sRMAddr, actor.ActorFunc(
 				func(context *actor.Context) error {
+					switch context.Message().(type) {
```
Comment (dzhu) [inline master/internal/api_user_intg_test.go:55]: Are the changes to this file supposed to be part of the PR?

Comment (NicholasBlaskey) [reply master/internal/api_user_intg_test.go:55]: Yeah, the experiment deletion flow needs the resource manager to respond to ```sproto.DeleteJob``` and calls something in ```trialLogBackend```

**Review by dzhu — APPROVED** [review]

**Pull request merged**